### PR TITLE
feat: add `dep check ./filesystem/folder` ability

### LINKED ISSUE #152: bb dep check should support local directory
@kuba-kubula found an interesting use case - he has a GitHub repo containing a package.json with dependencies, but the package is not published to any registry.

In this case, black belt cannot help him. It would be nice if it could work with local directory as well:

```
$ bb dep check react@1.2.3  # now
$ bb dep check ./my/local/path  # feature this issue is about
```

### PR DESCRIPTION
**Pull request by opichals**
Fixes #152

### CODE REVIEW

File: blackbelt/commands/dep.py [hunk 1]
```diff
@@ -13,9 +13,12 @@ def cli():
 
 
 def validate_dep(ctx, param, dep):
+    if os.path.isdir(dep): # check the provided project_dir
+        click.echo('The dependency is a local directory, using whatever is currently in node_modules')
+        if not os.path.isdir(os.path.join(dep, 'node_modules')):
+            raise click.BadParameter('There are no node_modules to analyze in the `'+dep+'` project directory')
```
Comment (honzajavorek) [inline blackbelt/commands/dep.py]: Install `flake8` or `pylama` 😉 Although, those would just notify you about the whitespace. Following is a proper solution to the problem:

```python
"There are no node_modules to analyze in the '{}' project directory".format(dep)
```

File: blackbelt/dependencies.py [hunk 2]
```diff
@@ -69,11 +69,8 @@ def check(dep, list_path, licenses_path, dev=False, debug=False):
     dep_name, dep_version = dep
     click.echo('Analyzing the package {0}@{1} ...'.format(dep_name, dep_version))
 
-    if dep_name == '.': # check the `pwd` project deps
-        click.echo('The dependency is a local directory, using whatever is currently in node_modules')
-        project_dir = '.'
-        if not os.path.isdir(os.path.join(project_dir, 'node_modules')):
-            raise click.BadParameter('There are no node_modules to analyze in the project directory')
+    if dep_version == '*': # check the `pwd` project deps
```
Comment (honzajavorek) [inline blackbelt/dependencies.py]: I'm not sure the comment ("check the `pwd` project deps") is still true if we added support for arbitrary directories

File: blackbelt/commands/dep.py [hunk 3]
```diff
@@ -13,9 +13,12 @@ def cli():
...
+        return (dep, '*')
```
Comment (honzajavorek) [inline blackbelt/commands/dep.py]: If I understand it correctly, the `*` is just a placeholder here to let the other part of the code know the first item in the tuple is a relative path? In that case I'd use `(dep, None)`, which doesn't look like a magical npm-specific syntax with some special meaning.

Beware, if you use that, then the comparison needs to be a bit different:

```python
if dep_version is None:
```

**Comment by opichals** [comment]
@honzajavorek Thanks, fixed the string formatting and adjusted to use the `dep_version = None`. 

Please have a look.

**Pull request merged**